### ISSUE TITLE
check if object writes are happening in parallel

### ISSUE THREAD

**Issue by inverted-capital** [dreamcatcher-tech/artifact#94]
Git might well be writing in parallel, in which case returning instantly from object writes won't help any, since we still need to await them all before doing the atomic commit part.

However, if they are not in parallel, significant speed can come from returning immediately and awaiting all promises to finish before moving on.

If any fail, bail on the commit.

**Comment by inverted-capital** [comment]
superseded by #101
